Include/internal/pycore_symtable.h

```diff
@@ -83,6 +107,7 @@ typedef struct _symtable_entry {
     PyObject *ste_directives;/* locations of global and nonlocal statements */
     PyObject *ste_mangled_names; /* set of names for which mangling should be applied */
     _Py_block_ty ste_type;
+    _extra_flags_t ste_extra_flags; /* extra temporary flags */
```

Comment:
What do you think of adding a `char *` here with the "scope name" instead of this flag? I feel like that would simplify things since you don't need to keep track of flag values.

If we go with that, we can also get rid of the enum entries for type aliases and type params, and simply pass a different scope name string when creating the symbol table entry.

Reply:
Mmh, if we are only using it for error messages, then I think it is a good idea. Should i actually remove everything in one PR or should I (for now) only focus on using the current interface + scope name without changing what you've written?

Reply:
Yes, the distinction between TypeVarBoundBlock, TypeAliasBlock, and TypeParamBlock is purely for error messages; they work the same. We can make all the changes in this PR.

Reply:
But what about the exported constants in the `symtable` module? and https://docs.python.org/3/library/symtable.html#symtable.SymbolTable.get_type? wouldn't it break some public API?

Reply:
The C API is private and we can change it, but you're right about the symtable module. I suppose "TypeVar bound" there is also incorrect.